core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -3966,23 +3966,23 @@ object Stream extends StreamLowPriority {
     * res0: List[Int] = List(0, 1, 2, 3, 4, 5)
     * }}}
     */
-  def unfoldLoop[F[x] <: Pure[x], S, O](s: S)(f: S => (O, Option[S])): Stream[F, O] =
-    Pull
-      .loop[F, O, S] { s =>
-        val (o, sOpt) = f(s)
-        Pull.output1(o) >> Pull.pure(sOpt)
-      }(s)
-      .stream
+  def unfoldLoop[F[x] <: Pure[x], S, O](start: S)(f: S => (O, Option[S])): Stream[F, O] = {
+    def recur(s: S): Pull[F, O, Unit] = f(s) match {
```

Comment:
I'd change all these `recur` to `loop` or `go`. Not sure which is more prevalent in the library but I don't think we use `recur` much of anywhere.

Reply:
Done.